services/slam/fake/slam.go

```diff
@@ -50,7 +84,7 @@ func init() {
 				config config.Service,
 				logger golog.Logger,
 			) (interface{}, error) {
-				return &SLAM{Name: config.Name, logger: logger}, nil
+				return &SLAM{Name: config.Name, logger: logger, dataCount: -1}, nil
```

Comment:
This & the new `count()` method fixes an off-by-one error where the first map dataset (which contains a `0` was never returned as the increment occurs before looking up the dataset).